modules/core/src/main/java/org/pixel/graphics/render/SpriteBatch.java

```diff
@@ -404,9 +404,9 @@ public void draw(Texture texture, Vector2 position, Rectangle source, Color colo
         spriteData.x = position.getX();
         spriteData.y = position.getY();
         spriteData.width =
-                texture.getWidth() * (source != null ? source.getWidth() / texture.getWidth() * scaleX : scaleX);
+                texture.getWidth() * (source != null ? Math.abs(source.getWidth()) / texture.getWidth() * scaleX : scaleX);
```

Comment:
why is this here? it should be added by the draw() caller as it impacts performance here.

Reply:
It seems that when I remove this I can't get tile flipping to work at all. I don't remember why I changed this but here is an example with Math.abs:
![image](https://user-images.githubusercontent.com/13765599/161380194-8bf38037-0ba6-40b5-b6d4-7d4def611791.png)


Without Math.abs
![image](https://user-images.githubusercontent.com/13765599/161380170-176b2f32-4e13-4a40-9c49-8020d8c97fea.png)

Reply:
Solved by using the display area parameter